js/core/webidl-oldschool.js

```diff
@@ -118,7 +118,7 @@ define(
         WebIDLProcessor.prototype = {
             setID:  function (obj, match) {
                 obj.id = match;
-                obj.refId = obj.id.replace(/[^a-zA-Z_\-]/g, "");
+                obj.refId = obj.id.replace(/[^a-zA-Z0-9_\-]/g, "");
```

Comment:
I don't think this is quite right yet; the id needs to not start with a number. I guess
`obj.refId = obj.id.replace(/^[^a-zA-Z_]*[^a-zA-Z0-9_\-]*/, "");`